### ISSUE TITLE
Abide Validation Radio Field required

### ISSUE THREAD

**Issue by designerno1** [zurb/foundation-sites#7196]
i got a form
```html
<form id="create_account" action="/create_account.php" method="post" data-abide novalidate>
  <div class="row" data-equalizer>
    <fieldset data-equalizer-watch>
      <legend>Some heading</legend>
      <input required type="radio" name="gender" value="m" id="93b5777073645a556be5027b4f5a9b08" />
      <label for="93b5777073645a556be5027b4f5a9b08">yes</label>
      <input required type="radio" name="gender" value="f" id="48b31e1ebc91b3271f4f267ffb83ab31" />
      <label for="48b31e1ebc91b3271f4f267ffb83ab31">no</label>
      *
    </fieldset>
  </div>
  <button class="button" type="submit" value="Submit">Submit</button>
</form>
```
the gender gets marked as error even if i check one radio

if i remove the "required" it doesn't get the error but doesn't even get's submitted

**Comment by zurbchris** [comment]
This one is tricky. I think for this use case the `required` attribute should go on the field set, and one of the elements within that fieldset should pass validation. Thoughts? Note, the feature I just described is not included in #7590 at the moment.

**Comment by designerno1** [comment]
i thought there was no problem with radio in my solution of abide #7470

**Comment by designerno1** [comment]
pattern on textarea does not work?

**Comment by FlorianBoe** [comment]
To open up a new discussion on this issue I looked into the specification.
W3C and WHATWG are quite specific on the use of required attributes on radio inputs:
> To avoid confusion as to whether a radio button group is required or not, authors are encouraged to specify the attribute on all the radio buttons in a group. Indeed, in general, authors are encouraged to avoid having radio button groups that do not have any initially checked controls in the first place, as this is a state that the user cannot return to, and is therefore generally considered a poor user interface.
> - https://www.w3.org/TR/html/forms.html#the-required-attribute

Handling this use case should be part of the solution.

**Comment by gakimball** [comment]
Fixed via cc2f2c668bb4f795cfad130915baf1b5f45fd79b. I wrote a test case [here](https://github.com/zurb/foundation-sites/blob/cc2f2c668bb4f795cfad130915baf1b5f45fd79b/test/visual/abide/abide-radio.html) with two Abide forms: one with a required radio group, and one with an optional radio group, showing both working as expected.

Regarding the W3C spec, to be nice to developers, a radio group is considered "required" if *any* of the `<input>`s in the group have the `required` attribute.

**Comment by Visiluna** [comment]
Then why am I still having this problem on 6.2?

**Comment by Visiluna** [comment]
I just pasted your test case into CodePen with Foundation 6 and the radio button group with required buttons does not work. I get the same result as I'm getting with mine. That is, the field won't validate, even when one of the radio options is selected. The non-required group works fine.

**Comment by kball** [comment]
@Visiluna the changeset in question was made Mar 1, 6.2 shipped Feb 26, so the fix was not in 6.2.  We're in final testing for a 6.2.1 release that includes this change, so we should have the fix live for you by end of week.

**Comment by briantan888** [comment]
Proposed pr #8542 for checkbox and radio group validation.
